src/periodic.jl

```diff
@@ -0,0 +1,85 @@
+const PeriodicFunction = Union{typeof(cos), typeof(sin)}
+const PeriodicParameter = Union{Real,Period}
+
+"""
+    Periodic(f, period, phase_shift) <: Transform
+
+Applies a periodic function `f` with provided `period` and `phase_shift` to the data.
+
+# Fields
+* `f::PeriodicFunction`: the periodic function
+* `period<:PeriodicParameter`: the duration it takes for the periodic function to repeat.
+    The sign of the value determines the direction of the function. Must be non-zero.
+* `phase_shift<:PeriodicParameter`: adjusts the phase of the periodic function, measured
+    in the same units as the input. Increasing the value translates the function to the
+    right, toward higher/later input values.
+"""
+struct Periodic{T<:PeriodicParameter} <: Transform
+    f::PeriodicFunction
+    period::T
+    phase_shift::T
+end
```

Comment:
perhaps define an inner constructor that checks that `period` is strictly positive?

Reply:
See other comment, but I've allowed `period` to be negative. The reason is that `FeatureEngineering.periodic` allows this. But it's only a convenience (you can negate the data or change `phase_shift` to get the same effect), and I don't think it matters from a feature engineering standpoint, so it could be dropped.

Reply:
>The reason is that FeatureEngineering.periodic allows this.

yeah and I'm equally confused about that. I don't see the justification for it especially since

>you can negate the data or change phase_shift to get the same effect

>I don't think it matters from a feature engineering standpoint, so it could be dropped.

I'd like to drop it tbh - if we discover a use case later that needs it we can add it back but I can't seem to find where it's used.